railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -141,6 +141,7 @@ def test
     end
 
     def tmp
+      empty_directory_with_keep_file "tmp"
```

Comment:
Do we need this if we are already recreating the directory on each rake restart run?.